### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by blacksheeep**
This pull request should fix some of the problematic typing problem and other potentional security fixes

### Contribution description
Fix, Type Confusions, Unsafe Function Calls,...

### CODE REVIEW

**Comment by mfrey** [comment]
You shouldn't update your feature branch by merging your master into the feature branch, but by using fetch/rebase. Doing otherwise is considered bad style.

**Comment by mfrey** [comment]
I suggest to make a clean new feature branch and cherry-pick the corresponding commits over to the new (clean) branch.

File: src/ccnl-core/include/ccnl-pkt-util.h [hunk 1]
```diff
@@ -22,8 +22,9 @@
 #ifndef CCNL_PKT_UTIL_H
 #define CCNL_PKT_UTIL_H
 
-#include <stddef.h>
+#include <stdint.h>
 #include <stdbool.h>
+#include <stddef.h>
 
 bool
 ccnl_isSuite(int suite);
```
Comment (blacksheeep) [inline src/ccnl-core/include/ccnl-pkt-util.h]: int?

File: src/ccnl-core/include/ccnl-pkt.h [hunk 2]
```diff
@@ -74,33 +77,34 @@ struct ccnl_pktdetail_ccntlv_s {
...
 #ifdef USE_HMAC256
-    unsigned char *hmacStart;
-    int hmacLen;
-    unsigned char *hmacSignature;
+    uint8_t *hmacStart;
+    size_t hmacLen;
+    uint8_t *hmacSignature;
 #endif
     unsigned int flags;
     char suite;
```
Comment (blacksheeep) [inline src/ccnl-core/include/ccnl-pkt.h]: unsigned?

File: src/ccnl-core/include/ccnl-prefix.h [hunk 3]
```diff
@@ -25,21 +25,22 @@
...
 struct ccnl_content_s;
 
 struct ccnl_prefix_s {
-    unsigned char **comp; /**< name components of the prefix without '\0' at the end */
-    int *complen; /**< length of the name components */
-    int compcnt; /**< number of name components */
+    uint8_t **comp; /**< name components of the prefix without '\0' at the end */
+    size_t *complen; /**< length of the name components */
+    uint32_t compcnt; /**< number of name components */
     char suite; /**< type of the packet format */
```
Comment (blacksheeep) [inline src/ccnl-core/include/ccnl-prefix.h]: unsigned

**Comment by blacksheeep** [comment]
Currently CCN-Lite is not C99 compiled, which causes an error when compiling for loops with iterator defined in the for loop

**Comment by blacksheeep** [comment]
> I suggest to make a clean new feature branch and cherry-pick the corresponding commits over to the new (clean) branch.

How important is this, in case we squash the commits anyway?

File: src/ccnl-core/include/ccnl-pkt-util.h (hunk 1 above)
Comment (mfrey) [reply src/ccnl-core/include/ccnl-pkt-util.h]: why? ``bool`` was introduced as a type in C99

Comment (blacksheeep) [reply src/ccnl-core/include/ccnl-pkt-util.h]: currently we have kernel support, therefore there is no support for C99

**Pull request closed without merging**